### PR TITLE
Do not remove the database if there is no backup and deal with manifest

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
In pg13, when creating a backup of the database, the pg_basebackup utility
will create a file called backup_manifest in a temporary file. smdba does
not take this into account, and when you run smdba backup-hot a second time,
smdba aborts because it finds this file already.

The issue is that smdba aborts the execution after having removed the database,
not before, so there is no backup to be restored.

This commit deals with backup_manifest file, and abort a restore if there
is no backup to be restored, before removing the actual database.

Signed-off-by: Jordi Massaguer Pla <jmassaguerpla@suse.de>

### CODE REVIEW

File: src/smdba/postgresqlgate.py
Comment (juliogonzalez) [inline src/smdba/postgresqlgate.py:608]: I see this is just a refactor, but find this naming really confusing.

We are trying to restore a backup from a file, but we use a variable called `destination_tar` :-?

**Review by juliogonzalez — APPROVED** [review]
I think this should work, but I suggest you get a second review from someone at ion squad... just in case.

File: src/smdba/postgresqlgate.py
Comment (brejoc) [inline src/smdba/postgresqlgate.py:832]: You could also use `os.path.join` here. But this would also work. It's just not as nice.

**Review by brejoc — APPROVED** [review]
Overall this looks good. Just the `os.path.join` change would be a nice to have.

File: src/smdba/postgresqlgate.py
Comment (jordimassaguerpla) [reply src/smdba/postgresqlgate.py:832]: I just copied what was one line above ;)

Comment (jordimassaguerpla) [reply src/smdba/postgresqlgate.py:832]: I am sure it would be better to have os.path.join, but then we would also need to change, in my opinion, the line above it, or it is weird to use different ways for the same thing (weird=makes it difficult to understand/read). I just found it weird to refactor existing/working code, so this is why I would prefer to leave it as it is. Sounds good?

Comment (brejoc) [reply src/smdba/postgresqlgate.py:832]: Sure, that's okay with me.

File: src/smdba/postgresqlgate.py
Comment (jordimassaguerpla) [reply src/smdba/postgresqlgate.py:608]: not nice, but tbh I don't want to include a name change refactor at this point. I just move this because I needed to add an "if" that needs this variable, nothing else.

**Pull request merged**
